book/website/reproducible-research/code-reuse/code-reuse-packages.md

```diff
@@ -0,0 +1,164 @@
+- {ref}`Version Control <rr-vcs>` of the codebase to track development and fix bugs.
+- {ref}`Documentation <rr-rdm-metadata>`. This might be in the form of a website or a wiki. There are even language specific packages that aim to provide project templates that will render the template into a website or other form of documentation (see [packagedown](https://pkgdown.r-lib.org/) for R, [Sphinx](https://www.sphinx-doc.org/en/master/) for Python or [Doxygen](https://www.doxygen.nl/) for C++).
+- Host your versioned codebase somewhere accessible. There are a whole range of places to host code depending on your use case. Public GitHub/GitLab repositories are extremely common especially in open source, where private repositories can be used for proprietary packages (think Matlab). Even just a local git repository might be enough for your use-case. 
+- Use Continuous Integration /  Continuous Development principles and pipelines such as GitHub Actions, Jenkins, Travis or GitLab Runners to help with a range of testing procedures.
+ - Consider submitting your package to a Central Resource such as those in the [](central-resources) above. There are usually strict requirements needed as your code is more likely to be distributed and used (more on this later on). Often GitHub repositories are used to store both stable and "daily"/"nightly" builds of a package in addition to a central resource so that users can submit bugs an contribute to the current version, while testing experimental versions as a preview. 
+ - Write a publication. This could be in the form of a a journal submission such as Numpy (in Nature!), or an abstract/paper in a conference proceeding. This allows you to peer review your package with experts in the field of intended use and might give more confidence to potential users that the package is robust.
+ - Publicizing your package. If you have gotten as far as hosting and publishing your package for others to use, you can publicize it in a variety of ways to let people know about it. It used to be that getting your package into a Central Resource provided ample opportunity for your package to be shared, but as these resources grow this is not necessarily the case anymore. Aside from publications in journals - use social media such as Twitter, Discord and Slack, Medium articles and postings in websites. 
+ - Build a community. Hopefully a few collaborators will have worked together to build a package - and by thinking about how the initial developers of the packages interact with future users you might be able to build a strong community that will help improve and maintain the package going forward.
+
+ ## Get Started - use a template
+With so many existing packages across many languages - it should be pretty easy to find good examples of packages and their published code on somewhere such as GitHub. 
```

Comment:
```suggestion
With so many existing packages across many languages, it should be pretty easy to find good examples of packages and their published code on software repositories such as GitHub.
```

Reply:
Should this be re-worded to suggest that the reader consult the repository of their favourite packages to see how they’re implemented?